js/piwik.js

```diff
@@ -4962,14 +5001,23 @@ if (typeof Piwik !== 'object') {
                 /**
                  * Set heartbeat (in seconds)
                  *
-                 * @param int minimumVisitLength
-                 * @param int heartBeatDelay
+                 * @param int heartBeatDelay Defaults to 15.
                  */
-                setHeartBeatTimer: function (minimumVisitLength, heartBeatDelay) {
-                    var now = new Date();
+                setHeartBeatTimer: function (heartBeatDelay) {
+                    configHeartBeatDelay = (heartBeatDelay || 15) * 1000;
 
-                    configMinimumVisitTime = now.getTime() + minimumVisitLength * 1000;
-                    configHeartBeatTimer = heartBeatDelay * 1000;
+                    // if a tracking request has already been set, start the heart beat timeout
+                    if (lastTrackerRequestTime !== null) {
+                        setUpHeartBeat();
+                    }
+                },
+
+                /**
+                 * Clear heartbeat.
+                 */
+                clearHeartBeat: function () {
```

Comment:
if we name the other method `enableHeartBeat` we could name this one `disableHeartBeat` or maybe we should use something like `startHeartBeat` and `stopHeartBeat`. Just some random thoughts